src/tss2-fapi/ifapi_curl.c

```diff
@@ -435,6 +436,12 @@ ifapi_get_curl_buffer(unsigned char * url, unsigned char ** buffer,
         goto out_easy_cleanup;
     }
 
+    curl_easy_getinfo(curl, CURLINFO_RESPONSE_CODE, &http_code);
+    if (http_code >= 400) {
```

Comment:
`curl_easy_perform` returns `CURLE_HTTP_RETURNED_ERROR` if the error code is `>= 400`, I don't see how this check adds anything?

See:
  - https://curl.se/libcurl/c/curl_easy_perform.html
  - https://curl.se/libcurl/c/libcurl-errors.html

Reply:
Hmmm... That's weird.
I have a system here where the CA is not online and I get this:
```
ERROR:fapi:src/tss2-fapi/ifapi_curl.c:193:ifapi_curl_verify_ek_cert() Failed to convert intermediate certificate to X509 format. (size=1245): (then the segfault)
```

AFAIK this should not happen if curl_easy_perform() returns an error.

With the patch, it does work.

Reply:
Sounds like a bug in curl, but sometimes we just gotta code around those :-p